feat: add --clear flag

advasileva opened the pull request:
According to #11

To add the flag, I needed to pass another (fourth) parameter to the constructor, but this code did not pass the linter. There was an idea to create a `Settings` class for `Speco` execution flags, but it looked redundant. As a result, I implemented a Builder pattern that builds `Speco` with the right flags (yes, we recently studied it at the university :)

advasileva commented:
@yegor256, please assign a reviewer for this PR

yegor256 commented:
@volodya-lombrozo pls, take a look

volodya-lombrozo commented on src/main/resources/org/eolang/speco/clear.xsl:
```diff
@@ -0,0 +1,36 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" id="SG" version="2.0">
```
@advasileva Could you provide tests for that change, please?

advasileva replied:
@volodya-lombrozo I understand correctly that it is better to implement this through `.yaml` configs with `xmir`?

volodya-lombrozo replied:
@advasileva Apparently, yes.

volodya-lombrozo reviewed (changes requested):
@advasileva About the builder pattern:
> [Builder](https://en.wikipedia.org/wiki/Builder_pattern). Terrible concept, since it encourages us to create and use big, complex objects. If you need a builder, there is already something wrong in your code. Refactor it so any object is easy to create through its constructors.

It's not my words :) I get it from that [link](https://www.yegor256.com/2016/02/03/design-patterns-and-anti-patterns.html).
___
There is a [way](https://www.yegor256.com/2016/04/19/object-must-not-be-configurable.html) how we can refactor the code in order to make it more maintainable.
I suppose we can create 3 different classes: `DefaultSpeco`, `EolangSpeco`, `ClearSpeco` - the names could be different of course, but I believe you get an idea.

advasileva commented:
@volodya-lombrozo I edited the PR (rebase, split into `*Speco` classes and add transformation test). Please, take a look

volodya-lombrozo reviewed (changes requested):
@advasileva I noticed that the refactoring has simplified the logic and made the code clearer, in my opinion, but we can simplify it even more. Overall, it's a good set of changes. Thank you.
___
PS: It would be nice to divide that PR into 2:
1. Add `--clear-xmir` option
2. Refactoring
However, I don't think we have to divide the current PR, it's only for the future.

advasileva commented on src/main/java/org/eolang/speco/Main.java:
```diff
@@ -61,9 +61,24 @@ public final class Main implements Callable<Integer> {
-        new Speco(this.input, this.output, this.eolang).exec();
+        Speco speco = new DefaultSpeco(this.input, this.output);
+        if (this.eolang) {
+            speco = new EolangSpeco(speco);
```
@volodya-lombrozo I don't think utility class is a good idea -- https://www.yegor256.com/2014/05/05/oop-alternative-to-utility-classes.html#utility-classes-are-evil :)

advasileva replied:
@volodya-lombrozo btw, the current design allows for changes in multiple dimensions (input/output format (`EolangSpeco`) and flags (`ClearXmirSpeco`), which can be combined. That is, it may become a problem in the future when new `*Speco` is added

advasileva commented on src/main/java/org/eolang/speco/ClearXmirSpeco.java:
```diff
@@ -0,0 +1,98 @@
+
+    @Override
+    public void exec() throws IOException {
+        Files.createDirectories(this.output());
```
@volodya-lombrozo Yes, it is, because this is the only way to link classes (that I know of). If we used class inheritance and redefined methods in the style of the Factory Method pattern. But because of the composition, we have to explicitly prescribe the method we use, otherwise `origin` will take its methods instead of our overrides.

For this reason, I don't like the code I wrote

advasileva commented on src/main/java/org/eolang/speco/Speco.java:
```diff
@@ -23,130 +23,60 @@
  */
-final class Speco {
-
+public interface Speco {
```
@volodya-lombrozo This is an attempt to make `Speco` classes reusable in multiple dimensions. In addition to the previous answer: I do not know how to use the method of the origin class calling overridden methods in composing classes (e.g. use `exec` of `DefaultSpeco` with `output` of `EolangSpeco`).

advasileva commented:
@volodya-lombrozo I took your code from #93, it really solves the problem with overriding methods. I also fixed tests and `qulice` checks. I added `toXml` method to the `Walk` interface to reduce duplication.

Thanks a lot for the example with the code, so it's clearer what was meant. Solving this problem with a single `Speco` interface was not the my best idea.

volodya-lombrozo commented on src/main/java/org/eolang/speco/XmirWalk.java:
```diff
@@ -0,0 +1,80 @@
+
+    @Override
+    public void exec() throws IOException {
+        Files.createDirectories(this.output);
```
@advasileva Technically, you still have code duplication between `XmirWalk` and `EoWalk`. But let's move on. For now, I just suggest to add puzzle for further refactoring. What do you think?

volodya-lombrozo replied:
@advasileva To any, it doesn't matter

advasileva replied:
@volodya-lombrozo done

volodya-lombrozo commented on src/main/java/org/eolang/speco/Speco.java:
```diff
@@ -24,129 +24,20 @@
+public interface Speco {
     /**
-     * Takes source codes on EO, converts to xmir and applies the AOI tool.
+     * Applies XSL-transformations to file.
```
@advasileva It looks like we have `xml` here, it's not a file.

advasileva replied:
@volodya-lombrozo fixed

volodya-lombrozo commented on src/main/java/org/eolang/speco/EoWalk.java:
```diff
@@ -0,0 +1,109 @@
+import org.objectionary.aoi.launch.LauncherKt;
+
+/**
+ * The interface encapsulating applying of specialization to EO.
```
@advasileva Looks like it's not an interface

advasileva replied:
@volodya-lombrozo fixed

volodya-lombrozo reviewed (approved):
@advasileva Looks good to me. Thanks.

@yegor256 Could you merge that changes, please?

advasileva commented:
@yegor256 try to merge it, please. Maybe problem with time are already fixed...

yegor256 commented:
@rultor merge

rultor commented:
> @rultor merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](https://www.rultor.com/t/30861-1475037869)

rultor commented:
> @rultor merge

@advasileva @yegor256 Oops, I failed. You can see the full log [here](https://www.rultor.com/t/30861-1475037869) (spent 2hr)

```
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.090 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.090 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.090 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.090 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.090 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.090 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@data
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/text()[1]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.091 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.092 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.092 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.092 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
00:11:03.405 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 9 shift(s) in 467ms
00:11:03.641 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 1 shift(s) in 236ms
7b49f70652b0693a7a74f37f578802e828634c72a6f5d9515e0c667e37b5f884
'pid' file is absent on the server after the end of operation; it seems that we didn't manage to start Docker container correctly
```

advasileva commented:
@yegor256 sorry, could you try merging again?..

yegor256 commented:
@rultor merge

rultor commented:
> @rultor merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](https://www.rultor.com/t/30861-1480719101)

rultor commented:
> @rultor merge

@advasileva @yegor256 Oops, I failed. You can see the full log [here](https://www.rultor.com/t/30861-1480719101) (spent 2hr)

```
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.864 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.864 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.864 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.864 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[1]/o[1]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@data
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[1]/o[2]/o[2]/text()[1]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.865 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.866 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.866 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]
Matches both "node()" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "node()" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.866 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]/@base
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.866 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]/@line
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:28.866 [main] WARN  com.jcabi.xml.ConsoleErrorListener MDC= - #warning(): Ambiguous rule match for /program/speco[1]/version[3]/o[1]/o[3]/o[1]/o[2]/o[1]/@pos
Matches both "@*" on line 109 of file:///org/eolang/speco/1-2-specialization.xsl
and "@*" on line 96 of file:///org/eolang/speco/1-2-specialization.xsl
09:48:29.169 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 9 shift(s) in 441ms
09:48:29.383 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 1 shift(s) in 214ms
c29a62e82ff4f5f1c0d615d4242d1eb07036844f7b7e779a95ce13b420f3f063
'pid' file is absent on the server after the end of operation; it seems that we didn't manage to start Docker container correctly
```

advasileva commented:
@yegor256 merge this one too, please

rultor commented:
> @rultor merge

@advasileva @yegor256 Oops, I failed. You can see the full log [here](https://www.rultor.com/t/30861-1490315762) (spent 3hr)

```
13:39:03.707 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 7 attribute(s) loaded ["Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Kotlin-Runtime-Component", "Kotlin-Version", "Manifest-Version", "Multi-Release"]
13:39:03.707 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 3 attribute(s) loaded ["Build-Jdk-Spec", "Created-By", "Manifest-Version"]
13:39:03.708 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 3 attribute(s) loaded ["Build-Jdk-Spec", "Created-By", "Manifest-Version"]
13:39:03.708 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 3 attribute(s) loaded ["Build-Jdk-Spec", "Created-By", "Manifest-Version"]
13:39:03.709 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 19 attribute(s) loaded ["Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.710 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 22 attribute(s) loaded ["Bnd-LastModified", "Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Manifest-Version", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version", "Tool"]
13:39:03.711 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 20 attribute(s) loaded ["Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Multi-Release", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.712 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 23 attribute(s) loaded ["Bnd-LastModified", "Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-Description", "Bundle-DocURL", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Manifest-Version", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version", "Tool"]
13:39:03.713 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 19 attribute(s) loaded ["Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.714 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 6 attribute(s) loaded ["Build-Jdk-Spec", "Created-By", "JCabi-Build", "JCabi-Date", "JCabi-Version", "Manifest-Version"]
13:39:03.715 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 16 attribute(s) loaded ["Bnd-LastModified", "Build-Jdk-Spec", "Bundle-Description", "Bundle-DocURL", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Created-By", "Export-Package", "Import-Package", "Manifest-Version", "Require-Capability", "Tool"]
13:39:03.715 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 3 attribute(s) loaded ["Build-Jdk-Spec", "Created-By", "Manifest-Version"]
13:39:03.716 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 18 attribute(s) loaded ["Bnd-LastModified", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Main-Class", "Manifest-Version", "Multi-Release", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version", "Tool"]
13:39:03.717 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 24 attribute(s) loaded ["Automatic-Module-Name", "Bnd-LastModified", "Build-Jdk-Spec", "Bundle-Description", "Bundle-DocURL", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Include-Resource", "Manifest-Version", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version", "Tool"]
13:39:03.718 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 20 attribute(s) loaded ["Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Provide-Capability", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.718 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 19 attribute(s) loaded ["Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.719 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 15 attribute(s) loaded ["Automatic-Module-Name", "Bnd-LastModified", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Require-Capability", "Tool"]
13:39:03.720 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 9 attribute(s) loaded ["Automatic-Module-Name", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Export-Package", "Import-Package", "Manifest-Version", "Require-Capability"]
13:39:03.720 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 13 attribute(s) loaded ["Build-Jdk-Spec", "Bundle-Description", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Main-Class", "Manifest-Version", "Multi-Release", "Tool"]
13:39:03.721 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 19 attribute(s) loaded ["Agent-Class", "Build-Jdk-Spec", "Bundle-Description", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Can-Redefine-Classes", "Can-Retransform-Classes", "Can-Set-Native-Method-Prefix", "Created-By", "Export-Package", "Import-Package", "Manifest-Version", "Multi-Release", "Premain-Class", "Require-Capability", "Tool"]
13:39:03.722 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 23 attribute(s) loaded ["Automatic-Module-Name", "Bnd-LastModified", "Build-Jdk-Spec", "Bundle-Description", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Created-By", "Dependencies", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version", "Tool"]
13:39:03.723 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 9 attribute(s) loaded ["Build-Jdk-Spec", "Created-By", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Manifest-Version", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.723 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 9 attribute(s) loaded ["Build-Jdk-Spec", "Created-By", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Manifest-Version", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.724 [main] DEBUG com.jcabi.manifests.Manifests MDC= - 20 attribute(s) loaded ["Build-Date", "Build-Revision", "Build-Time", "Built-By", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-SymbolicName", "Bundle-Version", "Created-By", "Export-Package", "Implementation-Title", "Implementation-Vendor", "Implementation-Version", "Import-Package", "Manifest-Version", "Provide-Capability", "Require-Capability", "Specification-Title", "Specification-Vendor", "Specification-Version"]
13:39:03.726 [main] INFO  com.jcabi.manifests.Manifests MDC= - 0 attributes loaded from 54 stream(s) in 677ms, 53 saved, 601 ignored: []
13:39:04.637 [main] DEBUG com.jcabi.xml.XSDDocument MDC= - com.sun.org.apache.xerces.internal.jaxp.validation.SimpleXMLSchema detected 0 error(s)
13:39:04.661 [main] DEBUG org.eolang.parser.Syntax MDC= - Input of 21 EO lines compiled
13:39:05.145 [main] WARN  com.yegor256.xsline.TrFast MDC= - XSL 'add-refs' took 348ms (over 100ms)
13:39:05.320 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 3 shift(s) in 589ms
13:39:05.433 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 1 shift(s) in 34ms
13:39:05.822 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 9 shift(s) in 387ms
13:39:05.911 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 1 shift(s) in 85ms
13:39:06.040 [main] DEBUG org.eolang.speco.SpecoEoTest MDC= - Started compilation for /tmp/junit18094921541946461752/output with test head:
before: |
  [] > first
    [] > boom
      QQ
      .io
      .stdout > @
        "First!\n"
  [] > 
13:46:48.559 [main] DEBUG org.eolang.speco.SpecoEoTest MDC= - Finished compilation for /tmp/junit18094921541946461752/output and got 14 lines of output
13:46:48.610 [main] DEBUG com.jcabi.xml.XSDDocument MDC= - com.sun.org.apache.xerces.internal.jaxp.validation.SimpleXMLSchema detected 0 error(s)
13:46:48.612 [main] DEBUG org.eolang.parser.Syntax MDC= - Input of 34 EO lines compiled
13:46:48.720 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 3 shift(s) in 93ms
13:46:48.751 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 1 shift(s) in 10ms
13:46:49.086 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 9 shift(s) in 335ms
13:46:49.326 [main] DEBUG com.yegor256.xsline.Xsline MDC= - Transformed XML through 1 shift(s) in 238ms
13:46:49.358 [main] DEBUG org.eolang.speco.SpecoEoTest MDC= - Started compilation for /tmp/junit2911772797203999187/output with test head:
before: |
  [s] > counter
    memory > k
      s
    [] > next
      seq > @
        ^
        .k
  
82915b0c9815521448732c86fceffcdae03fbe5095553d31c16aa69f9834a265
'pid' file is absent on the server after the end of operation; it seems that we didn't manage to start Docker container correctly
```

advasileva commented:
@volodya-lombrozo Сould you see why this test freezes and as a result the whole build cancels in time? Locally, the program from this test is successfully executed in 7 minutes.

advasileva commented:
@yegor256 try merge pls

yegor256 commented:
@rultor merge

rultor commented:
> @rultor merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](https://www.rultor.com/t/30861-1506733383)

0pdd commented:
@advasileva the puzzle [#109](https://github.com/objectionary/speco/issues/109) is still not solved.

rultor commented:
> @rultor merge

@yegor256 Done! FYI, the full log is [here](https://www.rultor.com/t/30861-1506733383) (took me 15min)
